packages/gp2-components/src/templates/OutputForm.tsx

```diff
@@ -4,33 +4,105 @@ import {
   FormCard,
   LabeledTextField,
   Form,
+  GlobeIcon,
+  LabeledDropdown,
+  pixels,
+  usePushFromHere,
 } from '@asap-hub/react-components';
 import { gp2 as gp2Routing } from '@asap-hub/routing';
-
+import { UrlExpression } from '@asap-hub/validation';
+import { css } from '@emotion/react';
 import { useState } from 'react';
+import { buttonWrapperStyle, mobileQuery } from '../layout';
+
 import { documentTypeMapper } from './CreateOutputPage';
 
+const { rem } = pixels;
+
+const footerStyles = css({
+  display: 'flex',
+  gap: rem(24),
+  justifyContent: 'flex-end',
+  [mobileQuery]: {
+    flexDirection: 'column-reverse',
+  },
+});
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: rem(32),
+});
 type OutputFormType = {
   createOutput: (
     payload: gp2Model.OutputPostRequest,
   ) => Promise<gp2Model.OutputResponse>;
   documentType: gp2Routing.OutputDocumentTypeParameter;
 };
+
 const OutputForm: React.FC<OutputFormType> = ({
   createOutput,
   documentType,
 }) => {
+  const historyPush = usePushFromHere();
   const [title, setTitle] = useState('');
+  const [link, setLink] = useState('');
+  const [type, setType] = useState('');
+  const [subtype, setSubtype] = useState('');
   const currentPayload = {
     title,
     documentType: documentTypeMapper[documentType],
+    link,
+    type: type ? (type as gp2Model.OutputType) : undefined,
+    subtype: subtype ? (subtype as gp2Model.OutputSubtype) : undefined,
```

Comment:
You should be able to remove this casting by using generics on the useState

Reply:
using generic it doesn't allow for the empty string